Composer/packages/client/src/pages/publish/BotStatusList.tsx

```diff
@@ -7,82 +7,55 @@ import moment from 'moment';
-  botPublishHistoryList: { projectId: string; publishHistory: { [key: string]: PublishResult[] } }[];
-  botPublishTypesList: { projectId: string; publishTypes: PublishType[] }[];
-  publishDisabled: boolean;
-  updateItems: (items: IBotStatus[]) => void;
-  updatePublishHistory: (items: PublishResult[], item: IBotStatus) => void;
-  updateSelectedBots: (items: IBotStatus[]) => void;
-  changePublishTarget: (PublishTarget: string, item: IBotStatus) => void;
-  onLogClick: (item: PublishResult) => void;
-  onRollbackClick: (selectedVersion: PublishResult, item: IBotStatus) => void;
-};
+export interface BotStatusListProps {
+  botStatusList: BotStatus[];
+  botPublishHistoryList: BotPublishHistory;
 
-export const BotStatusList: React.FC<IBotStatusListProps> = (props) => {
-  const {
-    projectId,
-    items,
-    botPublishHistoryList,
-    botPublishTypesList,
-    publishDisabled,
-    updateItems,
-    updatePublishHistory,
-    changePublishTarget,
-    updateSelectedBots,
-    onLogClick,
-    onRollbackClick,
-  } = props;
-  const [selectedBots, setSelectedBots] = useState<IBotStatus[]>([]);
-  const [showHistoryBots, setShowHistoryBots] = useState<string[]>([]);
+  /** When set to true, disable the checkbox. */
+  disabled: boolean;
+  managePublishProfile: (skillId: string) => void;
+  checkedIds: string[];
+  onCheck: (skillIds: string[]) => void;
+  changePublishTarget: (PublishTarget: string, item: BotStatus) => void;
+  onRollbackClick: (selectedVersion: PublishResult, item: BotStatus) => void;
+}
 
+export const BotStatusList: React.FC<BotStatusListProps> = ({
+  botStatusList,
+  botPublishHistoryList,
+  disabled,
+  checkedIds,
+  onCheck,
+  managePublishProfile,
+  changePublishTarget,
+  onRollbackClick,
+}) => {
+  const [expandedBotIds, setExpandedBotIds] = useState<{ [botId: string]: boolean }>({});
   const [currentSort, setSort] = useState({ key: 'Bot', descending: true });
-  const sortByName = (ev: React.MouseEvent<HTMLElement>, column: IColumn): void => {
-    if (column.isSorted) {
-      column.isSortedDescending = !column.isSortedDescending;
-      const newItems: IBotStatus[] = items.reverse();
-      updateItems(newItems);
-    }
-  };
-  const changeSelected = (item: IBotStatus, isChecked?: boolean) => {
-    let newSelectedBots: IBotStatus[];
-    if (isChecked) {
-      newSelectedBots = [...selectedBots, item];
-    } else {
-      newSelectedBots = selectedBots.filter((bot) => bot.id !== item.id);
-    }
-    setSelectedBots(newSelectedBots);
-    updateSelectedBots(newSelectedBots);
-  };
 
-  const publishTargetOptions = (item: IBotStatus): IDropdownOption[] => {
+  const displayedItems: BotStatus[] = useMemo(() => {
+    if (currentSort.key !== 'Bot') return botStatusList;
+    if (currentSort.descending) return botStatusList;
+    return botStatusList.slice().reverse();
+  }, [botStatusList, currentSort]);
+
+  const publishTargetOptions = (item: BotStatus): IDropdownOption[] => {
```

Comment:
either make this a useMemo or name it appropriately for a function: getPublishTargetOptions

Reply:
use `getPublishTargetOptions`